### PR TITLE
ci: combined build and deployment steps

### LINKED ISSUE #91: Combine continuous build and continous deployment scripts
Example: https://github.com/animo/agent-cli-page/blob/main/.github/workflows/deploy.yaml

### PR DESCRIPTION
**Pull request by janrtvld**
Signed-off-by: Jan Rietveld [jan@animo.id](mailto:jan@animo.id)

### CODE REVIEW

**Comment by TimoGlastra** [comment]
We should integrate the build option for workflow dispatch:

```
  workflow_dispatch:
    inputs:
      build:
        default: true
        type: boolean
        required: false
        description: Build the website before deploying
```

and

```
        if: inputs.build == true
```

**Comment by janrtvld** [comment]
Is this the correct syntax? 
```yaml    
if: inputs.build == true && github.ref == 'refs/heads/main'`? 
```

And will this only check the `needs` if build is true? 
```yaml    
if: inputs.build == true && github.ref == 'refs/heads/main'
needs: [build-and-push-image-server, build-and-push-image-client]
```

File: .github/workflows/continuous-deployment.yml [hunk 1]
```diff
@@ -107,11 +110,9 @@ jobs:
 
   deploy:
     # Only run on main branch
-    if: github.ref == 'refs/heads/main'
     runs-on: ubuntu-latest
-    needs:
-      - build-and-push-image-server
-      - build-and-push-image-client
+    if: inputs.build == true && github.ref == 'refs/heads/main'
```
Comment (TimoGlastra) [inline .github/workflows/continuous-deployment.yml]: We always want to run the deploy script. this allows us to deploy without building first

**Comment by TimoGlastra** [comment]
> And will this only check the needs if build is true?

Good question.  I think it will work ?? ❓

**Review by TimoGlastra — APPROVED** [review]

**Pull request merged**